packages/arb-token-bridge-ui/src/hooks/useArbTokenBridge.ts

```diff
@@ -144,8 +145,20 @@ export const useArbTokenBridge = (
   // this prevents previous account/chains' transactions to show up in the current account
   // also makes sure the state of app doesn't get incrementally bloated with all accounts' txns loaded up till date
   useEffect(() => {
-    setPendingWithdrawalMap({})
-  }, [l1.provider, l2.provider, walletAddress])
+    if (!connector) {
+      return
+    }
+
+    const resetPendingWithdrawalMap = () => {
+      setPendingWithdrawalMap({})
+    }
+
+    connector.on('change', resetPendingWithdrawalMap)
```

Comment:
I would maybe just add a comment linking to https://github.com/wagmi-dev/wagmi/blob/main/packages/connectors/src/base.ts#L8 and saying this will trigger either when network or account changes.